watcher/watcher.go

```diff
@@ -14,18 +15,26 @@ import (
 	"github.com/kubescape/k8s-interface/k8sinterface"
 	"github.com/kubescape/k8s-interface/workloadinterface"
 	"github.com/kubescape/operator/utils"
+	spdxv1beta1 "github.com/kubescape/storage/pkg/apis/softwarecomposition/v1beta1"
+	kssc "github.com/kubescape/storage/pkg/generated/clientset/versioned"
 	"golang.org/x/exp/slices"
 	core1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
 )
 
+var (
+	ErrUnsupportedObject = errors.New("Unsupported object type")
+	ErrUnknownImageID    = errors.New("Unknown image ID")
+)
```

Comment:
thought: we should ideally make these custom error types, so we can store information inside them

Like image IDs, the type of the unsupported object and maybe its payload.